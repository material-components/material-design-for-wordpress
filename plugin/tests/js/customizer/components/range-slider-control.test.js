/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';
import { mount } from 'enzyme';

/**
 * Internal dependencies
 */
import RangeSliderControl from '../../../../assets/src/customizer/components/range-slider-control';

/**
 * Render the component.
 *
 * @param {Object} props - Component props
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <RangeSliderControl { ...props } /> );
};

/**
 * Mount the component.
 *
 * @param {Object} props - Component props
 * @return {Function} A functional component.
 */
const setupMount = props => {
	return mount( <RangeSliderControl { ...props } /> );
};

const mockOnChangeFn = jest.fn();
const mockOnExpandedFn = jest.fn();

const baseProps = {
	id: 'material-design-small_component_radius',
	label: 'Small Component Radius',
	description: 'This is a test description',
	value: 10,
	min: 0,
	max: 36,
	step: 1,
	onChange: mockOnChangeFn,
};

describe( 'RangeSliderControl', () => {
	beforeEach( () => {
		jest.clearAllMocks();
	} );

	it( 'matches snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when no description is provided', () => {
		const props = { ...baseProps };
		props.description = '';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when is global', () => {
		const props = { ...baseProps };
		props.isGlobal = true;
		props.expandedSettings = true;
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should invoke handleExpandedSettings when expand settings button is clicked', async () => {
		const props = { ...baseProps };
		props.isGlobal = true;
		props.expandedSettings = true;
		props.handleExpandedSettings = mockOnExpandedFn;
		const { container } = setup( props );

		const input = container.querySelector(
			'.range-slider-control-settings-expanded'
		);
		await fireEvent.click( input );

		expect( mockOnExpandedFn ).toHaveBeenCalledTimes( 1 );
	} );

	it( 'should have its value changed after the input number field value is changed', () => {
		const wrapper = setupMount( baseProps );

		const input = wrapper.find( 'input.components-range-control__number' );

		input.simulate( 'change', {
			target: {
				value: 5,
				checkValidity: () => true,
			},
		} );

		expect(
			wrapper.find( 'input.components-range-control__number' ).prop( 'value' )
		).toBe( 5 );

		expect( mockOnChangeFn ).toHaveBeenCalledTimes( 1 );
	} );

	it( 'should have its value changed after the input range field value is changed', () => {
		const wrapper = setupMount( baseProps );

		const input = wrapper.find( 'input.components-range-control__slider' );

		input.simulate( 'change', {
			target: {
				value: 4,
				checkValidity: () => true,
			},
		} );

		expect(
			wrapper.find( 'input.components-range-control__slider' ).prop( 'value' )
		).toBe( 4 );

		expect( mockOnChangeFn ).toHaveBeenCalledTimes( 1 );
	} );

	it( 'should show the description when the title is clicked', () => {
		const wrapper = setupMount( baseProps );

		wrapper.find( '.range-slider-control-title' ).simulate( 'click' );

		expect(
			wrapper.find( '.customize-control-description' ).getDOMNode()
		).toBeVisible();
	} );
} );
