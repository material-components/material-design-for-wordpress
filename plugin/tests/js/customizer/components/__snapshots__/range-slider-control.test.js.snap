// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSliderControl matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="range-slider-control"
        id="range-slider-control-material-design-small_component_radius"
      >
        <div
          class="range-slider-control-header"
        >
          <div
            class="range-slider-control-title range-slider-control-title-with-description"
            role="link"
            tabindex="0"
          >
            <span
              class="customize-control-title range-slider-control-title__item"
            >
              Small Component Radius
            </span>
            <i
              class="material-icons range-slider-control-title__item info-icon"
              title="More info"
            >
              info
            </i>
          </div>
          <span
            class="customize-control-description"
          >
            This is a test description
          </span>
        </div>
        <div
          class="range-slider-control-body"
        >
          <span
            class="range-slider-control-body__item slider"
          >
            <div
              class="components-base-control components-range-control"
            >
              <div
                class="components-base-control__field"
              >
                <span
                  class="components-range-control__root css-yaebbd-Root exqw8y20"
                >
                  <span
                    class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                    color="#00669b"
                  >
                    <input
                      aria-hidden="false"
                      class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                      id="inspector-range-control-0"
                      max="36"
                      min="0"
                      step="1"
                      tabindex="0"
                      type="range"
                      value="10"
                    />
                    <span
                      aria-hidden="true"
                      class="css-1lwhlrq-Rail exqw8y24"
                    />
                    <span
                      aria-hidden="true"
                      class="components-range-control__track css-ogrpie-Track exqw8y25"
                      style="width: 27.77777777777778%;"
                    />
                    <span
                      class="css-kjao9s-ThumbWrapper exqw8y29"
                      style="left: 27.77777777777778%;"
                    >
                      <span
                        aria-hidden="true"
                        class="css-c88ajq-Thumb exqw8y210"
                      />
                    </span>
                    <span
                      aria-hidden="false"
                      class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                      role="tooltip"
                      style="left: 27.77777777777778%; z-index: 100;"
                    >
                      10
                    </span>
                  </span>
                  <input
                    class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                    inputmode="decimal"
                    max="36"
                    min="0"
                    step="1"
                    type="number"
                    value="10"
                  />
                </span>
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="range-slider-control"
      id="range-slider-control-material-design-small_component_radius"
    >
      <div
        class="range-slider-control-header"
      >
        <div
          class="range-slider-control-title range-slider-control-title-with-description"
          role="link"
          tabindex="0"
        >
          <span
            class="customize-control-title range-slider-control-title__item"
          >
            Small Component Radius
          </span>
          <i
            class="material-icons range-slider-control-title__item info-icon"
            title="More info"
          >
            info
          </i>
        </div>
        <span
          class="customize-control-description"
        >
          This is a test description
        </span>
      </div>
      <div
        class="range-slider-control-body"
      >
        <span
          class="range-slider-control-body__item slider"
        >
          <div
            class="components-base-control components-range-control"
          >
            <div
              class="components-base-control__field"
            >
              <span
                class="components-range-control__root css-yaebbd-Root exqw8y20"
              >
                <span
                  class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                  color="#00669b"
                >
                  <input
                    aria-hidden="false"
                    class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                    id="inspector-range-control-0"
                    max="36"
                    min="0"
                    step="1"
                    tabindex="0"
                    type="range"
                    value="10"
                  />
                  <span
                    aria-hidden="true"
                    class="css-1lwhlrq-Rail exqw8y24"
                  />
                  <span
                    aria-hidden="true"
                    class="components-range-control__track css-ogrpie-Track exqw8y25"
                    style="width: 27.77777777777778%;"
                  />
                  <span
                    class="css-kjao9s-ThumbWrapper exqw8y29"
                    style="left: 27.77777777777778%;"
                  >
                    <span
                      aria-hidden="true"
                      class="css-c88ajq-Thumb exqw8y210"
                    />
                  </span>
                  <span
                    aria-hidden="false"
                    class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                    role="tooltip"
                    style="left: 27.77777777777778%; z-index: 100;"
                  >
                    10
                  </span>
                </span>
                <input
                  class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                  inputmode="decimal"
                  max="36"
                  min="0"
                  step="1"
                  type="number"
                  value="10"
                />
              </span>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RangeSliderControl matches snapshot when is global 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="range-slider-control"
        id="range-slider-control-material-design-small_component_radius"
      >
        <div
          class="range-slider-control-header"
        >
          <div
            class="range-slider-control-title range-slider-control-title-with-description"
            role="link"
            tabindex="0"
          >
            <span
              class="customize-control-title range-slider-control-title__item"
            >
              Small Component Radius
            </span>
            <i
              class="material-icons range-slider-control-title__item info-icon"
              title="More info"
            >
              info
            </i>
          </div>
          <span
            class="customize-control-description"
          >
            This is a test description
          </span>
        </div>
        <div
          class="range-slider-control-body"
        >
          <span
            class="range-slider-control-body__item slider"
          >
            <div
              class="components-base-control components-range-control"
            >
              <div
                class="components-base-control__field"
              >
                <span
                  class="components-range-control__root css-yaebbd-Root exqw8y20"
                >
                  <span
                    class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                    color="#00669b"
                  >
                    <input
                      aria-hidden="false"
                      class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                      id="inspector-range-control-2"
                      max="36"
                      min="0"
                      step="1"
                      tabindex="0"
                      type="range"
                      value="10"
                    />
                    <span
                      aria-hidden="true"
                      class="css-1lwhlrq-Rail exqw8y24"
                    />
                    <span
                      aria-hidden="true"
                      class="components-range-control__track css-ogrpie-Track exqw8y25"
                      style="width: 27.77777777777778%;"
                    />
                    <span
                      class="css-kjao9s-ThumbWrapper exqw8y29"
                      style="left: 27.77777777777778%;"
                    >
                      <span
                        aria-hidden="true"
                        class="css-c88ajq-Thumb exqw8y210"
                      />
                    </span>
                    <span
                      aria-hidden="false"
                      class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                      role="tooltip"
                      style="left: 27.77777777777778%; z-index: 100;"
                    >
                      10
                    </span>
                  </span>
                  <input
                    class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                    inputmode="decimal"
                    max="36"
                    min="0"
                    step="1"
                    type="number"
                    value="10"
                  />
                </span>
              </div>
            </div>
          </span>
          <span
            class="range-slider-control-body__item"
          >
            <button
              aria-label="View individual components"
              aria-pressed="true"
              class="components-button components-button is-link control-settings-expanded range-slider-control-settings-expanded is-pressed is-pressed is-link"
              type="button"
            >
              <span
                class="dashicons dashicons-admin-settings"
              />
            </button>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="range-slider-control"
      id="range-slider-control-material-design-small_component_radius"
    >
      <div
        class="range-slider-control-header"
      >
        <div
          class="range-slider-control-title range-slider-control-title-with-description"
          role="link"
          tabindex="0"
        >
          <span
            class="customize-control-title range-slider-control-title__item"
          >
            Small Component Radius
          </span>
          <i
            class="material-icons range-slider-control-title__item info-icon"
            title="More info"
          >
            info
          </i>
        </div>
        <span
          class="customize-control-description"
        >
          This is a test description
        </span>
      </div>
      <div
        class="range-slider-control-body"
      >
        <span
          class="range-slider-control-body__item slider"
        >
          <div
            class="components-base-control components-range-control"
          >
            <div
              class="components-base-control__field"
            >
              <span
                class="components-range-control__root css-yaebbd-Root exqw8y20"
              >
                <span
                  class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                  color="#00669b"
                >
                  <input
                    aria-hidden="false"
                    class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                    id="inspector-range-control-2"
                    max="36"
                    min="0"
                    step="1"
                    tabindex="0"
                    type="range"
                    value="10"
                  />
                  <span
                    aria-hidden="true"
                    class="css-1lwhlrq-Rail exqw8y24"
                  />
                  <span
                    aria-hidden="true"
                    class="components-range-control__track css-ogrpie-Track exqw8y25"
                    style="width: 27.77777777777778%;"
                  />
                  <span
                    class="css-kjao9s-ThumbWrapper exqw8y29"
                    style="left: 27.77777777777778%;"
                  >
                    <span
                      aria-hidden="true"
                      class="css-c88ajq-Thumb exqw8y210"
                    />
                  </span>
                  <span
                    aria-hidden="false"
                    class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                    role="tooltip"
                    style="left: 27.77777777777778%; z-index: 100;"
                  >
                    10
                  </span>
                </span>
                <input
                  class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                  inputmode="decimal"
                  max="36"
                  min="0"
                  step="1"
                  type="number"
                  value="10"
                />
              </span>
            </div>
          </div>
        </span>
        <span
          class="range-slider-control-body__item"
        >
          <button
            aria-label="View individual components"
            aria-pressed="true"
            class="components-button components-button is-link control-settings-expanded range-slider-control-settings-expanded is-pressed is-pressed is-link"
            type="button"
          >
            <span
              class="dashicons dashicons-admin-settings"
            />
          </button>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RangeSliderControl matches snapshot when no description is provided 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="range-slider-control"
        id="range-slider-control-material-design-small_component_radius"
      >
        <div
          class="range-slider-control-header"
        >
          <div
            class="range-slider-control-title"
            role="link"
            tabindex="0"
          >
            <span
              class="customize-control-title range-slider-control-title__item"
            >
              Small Component Radius
            </span>
          </div>
        </div>
        <div
          class="range-slider-control-body"
        >
          <span
            class="range-slider-control-body__item slider"
          >
            <div
              class="components-base-control components-range-control"
            >
              <div
                class="components-base-control__field"
              >
                <span
                  class="components-range-control__root css-yaebbd-Root exqw8y20"
                >
                  <span
                    class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                    color="#00669b"
                  >
                    <input
                      aria-hidden="false"
                      class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                      id="inspector-range-control-1"
                      max="36"
                      min="0"
                      step="1"
                      tabindex="0"
                      type="range"
                      value="10"
                    />
                    <span
                      aria-hidden="true"
                      class="css-1lwhlrq-Rail exqw8y24"
                    />
                    <span
                      aria-hidden="true"
                      class="components-range-control__track css-ogrpie-Track exqw8y25"
                      style="width: 27.77777777777778%;"
                    />
                    <span
                      class="css-kjao9s-ThumbWrapper exqw8y29"
                      style="left: 27.77777777777778%;"
                    >
                      <span
                        aria-hidden="true"
                        class="css-c88ajq-Thumb exqw8y210"
                      />
                    </span>
                    <span
                      aria-hidden="false"
                      class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                      role="tooltip"
                      style="left: 27.77777777777778%; z-index: 100;"
                    >
                      10
                    </span>
                  </span>
                  <input
                    class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                    inputmode="decimal"
                    max="36"
                    min="0"
                    step="1"
                    type="number"
                    value="10"
                  />
                </span>
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="range-slider-control"
      id="range-slider-control-material-design-small_component_radius"
    >
      <div
        class="range-slider-control-header"
      >
        <div
          class="range-slider-control-title"
          role="link"
          tabindex="0"
        >
          <span
            class="customize-control-title range-slider-control-title__item"
          >
            Small Component Radius
          </span>
        </div>
      </div>
      <div
        class="range-slider-control-body"
      >
        <span
          class="range-slider-control-body__item slider"
        >
          <div
            class="components-base-control components-range-control"
          >
            <div
              class="components-base-control__field"
            >
              <span
                class="components-range-control__root css-yaebbd-Root exqw8y20"
              >
                <span
                  class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                  color="#00669b"
                >
                  <input
                    aria-hidden="false"
                    class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                    id="inspector-range-control-1"
                    max="36"
                    min="0"
                    step="1"
                    tabindex="0"
                    type="range"
                    value="10"
                  />
                  <span
                    aria-hidden="true"
                    class="css-1lwhlrq-Rail exqw8y24"
                  />
                  <span
                    aria-hidden="true"
                    class="components-range-control__track css-ogrpie-Track exqw8y25"
                    style="width: 27.77777777777778%;"
                  />
                  <span
                    class="css-kjao9s-ThumbWrapper exqw8y29"
                    style="left: 27.77777777777778%;"
                  >
                    <span
                      aria-hidden="true"
                      class="css-c88ajq-Thumb exqw8y210"
                    />
                  </span>
                  <span
                    aria-hidden="false"
                    class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                    role="tooltip"
                    style="left: 27.77777777777778%; z-index: 100;"
                  >
                    10
                  </span>
                </span>
                <input
                  class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                  inputmode="decimal"
                  max="36"
                  min="0"
                  step="1"
                  type="number"
                  value="10"
                />
              </span>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
